src/groupeddataframe/grouping.jl

```diff
@@ -373,17 +374,53 @@ function combine(f::Any, gd::GroupedDataFrame)
         return gd.parent[1:0, gd.cols]
     end
 end
+# ^ old syntax with a do call
+# old syntax
 combine(gd::GroupedDataFrame, f::Any) = combine(f, gd)
-combine(gd::GroupedDataFrame, f::Pair...) = combine(f, gd)
-combine(gd::GroupedDataFrame, f::Pair) = combine(f, gd)
+# pair syntax  
+function combine(gd::GroupedDataFrame, args::Union{Pair, AbstractVector{<:Pair}}...; kwargs...)
 
-function combine(gd::GroupedDataFrame; f...)
-    if length(f) == 0
-        Base.depwarn("combine(gd) is deprecated, use DataFrame(gd) instead", :combine)
-        combine(identity, gd)
-    else
-        combine(values(f), gd)
+    # handle pairs on their own so we can iterate 
```

Comment:
That's really too much code for a small feature IMHO. Is there any way to make this simpler? We don't have to allow mixing keyword and positional arguments.